util/capacity-reservation.py

```diff
@@ -58,20 +62,57 @@ def parse_args():
     return parser.parse_args()
 
 
-def create_reservation(client, instance_type, zone, count, tag_path, dry_run):
+def _retrieve_tags(client, instance_id):
+    """Retrieve tags from an EC2 instance"""
+    response = client.describe_tags(
+        Filters=[
+            {
+                'Name': 'resource-id',
+                'Values': [instance_id]
+            }
+        ]
+    )
+    if 'Tags' in response:
+        tags = [
+            {'Value': tag['Value'], 'Key': tag['Key']}
```

Comment:
a lot going on in this list comprehension for which i don't quite comprehend but i think it looks ok :)